packages/datasets/src/bin/huggingFace/uploadCodeExampleDatasetToHuggingFace.ts

```diff
@@ -0,0 +1,121 @@
+    MONGODB_CONNECTION_URI,
+    MONGODB_DATABASE_NAME,
+  } = assertEnvVars({ ...HUGGINGFACE, ...CORE_CHATBOT_APP_ENV_VARS });
+
+  const pageStore = makeMongoDbPageStore({
+    connectionUri: MONGODB_CONNECTION_URI,
+    databaseName: MONGODB_DATABASE_NAME,
+  });
+
+  const transformedContentStore =
+    makeMongoDbTransformedContentStore<CodeExampleDatasetEntry>({
+      connectionUri: MONGODB_CONNECTION_URI,
+      databaseName: MONGODB_DATABASE_NAME,
+      collectionName: "code_examples",
+    });
+
+  const MAX_PAGE_TRANSFORM_CONCURRENCY = 8;
+
+  const transformPage = await makeTranformPageToAnnotatedCodeExamples(
+    openAiClient,
+    model,
+    MAX_PAGE_TRANSFORM_CONCURRENCY
+  );
+  try {
+    const pages = await pageStore.loadPages({
+      query: makeLoadPagesFilter(
+        publicDatasetSourceName,
+        Array.from(forbiddenUrls)
+      ),
+    });
+    logger.info(`Loaded pages from MongoDB. Loaded ${pages.length} pages.`);
+
+    // Only transforms content when page was updated after the last transformed content update
+    await updateTransformedContent({
+      pages,
+      transformedContentStore,
+      transformPage,
+    });
+    logger.info(`Updated transformed content in MongoDB`);
+
+    // Get entire code example dataset
+    const fullCodeExampleDataset = await transformedContentStore.loadContent();
+    logger.info(
+      `Loaded code example dataset from MongoDB. Loaded ${fullCodeExampleDataset.length} code examples.`
+    );
+
+    const currentDate = new Date().toISOString();
+    // Date formatted as yyyy-mm-dd
+    const yyyyMmDd = currentDate.split("T")[0];
+
+    const fileBaseName = "public-code-examples";
+
+    logger.info(
+      `Uploading dataset to Hugging Face repo '${HUGGINGFACE_DOCS_CODE_EXAMPLE_REPO}'`
+    );
+
+    const res = await uploadDatasetToHuggingFace({
+      huggingFace: {
+        repoName: HUGGINGFACE_DOCS_CODE_EXAMPLE_REPO,
+        accessToken: HUGGINGFACE_ACCESS_TOKEN,
+      },
+      commit: {
+        title: `Dataset upload ${yyyyMmDd}`,
+        description: `Public MongoDB code example dataset upload ${currentDate}`,
+      },
+      data: [
+        {
+          path: path.format({
+            name: fileBaseName,
+            ext: ".json",
+          }),
+          content: new Blob([JSON.stringify(fullCodeExampleDataset)]),
+        },
+      ],
+    });
+    logger.info(
+      `Uploaded dataset to Hugging Face repo '${HUGGINGFACE_DOCS_CODE_EXAMPLE_REPO}'`
+    );
+    logger.info(res);
+  } finally {
+    await Promise.allSettled([
+      pageStore.close(),
+      transformedContentStore.close(),
+    ]);
+    logger.info("Closed MongoDB connection(s)");
+  }
+}
+uploadCodeExampleDatasetToHuggingFace();
+function makeLoadPagesFilter(
+  publicDatasetSourceName: RegExp,
+  arg1: string[]
+):
+  | import("mongodb").Filter<import("mongodb-rag-core").PersistedPage>
+  | undefined {
+  throw new Error("Function not implemented.");
+}
```

Comment:
Something weird going on here

Reply:
yea looks machine generated. will investigate and fix